tests/loom.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg(loom)]
+
+use loom::sync::{mpsc, Arc};
+use loom::thread;
+
+use async_lock::Barrier;
+
+#[ignore]
```

Comment:
This PR doesn't seem to add any tests that are actually run in CI.

Reply:
Still WIP here.